### PR TITLE
Ariane: skip E2E tests when changing unit tests only

### PR DESCRIPTION
**Pull request by giorio94**
Please see the commit messages for additional details.

### CODE REVIEW

**Review by aanm — APPROVED** [review]
Should we backport this?

File: .github/ariane-config.yaml [hunk 1]
```diff
@@ -97,42 +97,42 @@ triggers:
...
   conformance-gke.yaml:
-    paths-ignore-regex: (bpf/complexity-tests|bpf/tests|test|Documentation|hubble|CODEOWNERS|USERS.md)/
+    paths-ignore-regex: ((bpf/complexity-tests|bpf/tests|test|Documentation|hubble)/|(CODEOWNERS|USERS.md|.*_test\.go)$)
   conformance-ingress.yaml:
-    paths-ignore-regex: (bpf/complexity-tests|bpf/tests|test|Documentation|hubble|CODEOWNERS|USERS.md)/
+    paths-ignore-regex: ((bpf/complexity-tests|bpf/tests|test|Documentation|hubble)/|(CODEOWNERS|USERS.md|.*_test\.go)$)
   conformance-multi-pool.yaml:
-    paths-ignore-regex: (bpf/complexity-tests|bpf/tests|test|Documentation|hubble|CODEOWNERS|USERS.md)/
+    paths-ignore-regex: ((bpf/complexity-tests|bpf/tests|test|Documentation|hubble)/|(CODEOWNERS|USERS.md|.*_test\.go)$)
   conformance-runtime.yaml:
```
Comment (gandro) [inline .github/ariane-config.yaml:123]: Don't the runtime tests include a run of privileged tests stored in `_test.go`?

Comment (gandro) [reply .github/ariane-config.yaml:123]: https://github.com/cilium/cilium/blob/5d0e9d3933c9fc90a951ef385f2ea58c609eb1f0/.github/workflows/conformance-runtime.yaml#L379-L395

Comment (giorio94) [reply .github/ariane-config.yaml:123]: Yes, and indeed they are not skipped in that case:

```yaml
conformance-runtime.yaml:
    paths-ignore-regex: ((bpf/complexity-tests|bpf/tests|cilium-cli|Documentation)/|(CODEOWNERS|USERS.md)$)
```

**Comment by giorio94** [comment]
> Should we backport this?

I would say it doesn't hurt, although it is probably more unlikely to have PRs not touching code files as well on stable branches. That said, previous PRs modifying this config don't appear to have been backported, hence it would need to be a manual backport to essentially re-sync the configs.

File: .github/ariane-config.yaml (hunk 1 above)
Comment (gandro) [reply .github/ariane-config.yaml:123]: :facepalm: Sorry for the noise, I slipped in the line and looked at multi-pool

**Pull request merged**